File: output/python_rest_sdk/client_sdk/models/new_pet.py
""" """

from dataclasses import dataclass, field
from typing import Optional, List, Dict, Any, Union
from datetime import datetime, date


@dataclass
class Newpet:
    """ """

    name: String  # Name of the pet
    tag: Optional[String] = None  # Tag associated with the pet
    photo_urls: Optional[String] = None  # URLs of pet photos

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> "Newpet":
        """Create an instance from a dictionary."""
        return cls(
            name=data.get("name"), tag=data.get("tag"), photo_urls=data.get("photoUrls")
        )

    def to_dict(self) -> Dict[str, Any]:
        """Convert to a dictionary."""
        result = {}
        if self.name is not None:
            result["name"] = self.name
        if self.tag is not None:
            result["tag"] = self.tag
        if self.photo_urls is not None:
            result["photoUrls"] = self.photo_urls
        return result

    def __str__(self) -> str:
        """String representation."""
        return f"Newpet(name={self.name}, tag={self.tag}, photo_urls={self.photo_urls})"

File: output/rust_rest_sdk/src/models/new_pet.rs
//! 
//! 

use serde::{Deserialize, Serialize};
use std::collections::HashMap;
use chrono::{DateTime, NaiveDate, Utc};

/// 
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct Newpet {
    /// Name of the pet
    #[serde(rename = "name")]
    pub name: String,
    
    /// Tag associated with the pet
    #[serde(skip_serializing_if = "Option::is_none")]
    #[serde(rename = "tag")]
    pub tag: Option<String>,
    
    /// URLs of pet photos
    #[serde(skip_serializing_if = "Option::is_none")]
    #[serde(rename = "photoUrls")]
    pub photo_urls: Option<String>,
    
}

impl Newpet {
    /// Create a new instance with required fields
    pub fn new(name: String, ) -> Self {
        Self {
            name,
            tag: None,
            photo_urls: None,
        }
    }
    
    
    /// Set Tag associated with the pet
    pub fn with_tag(mut self, tag: String) -> Self {
        self.tag = Some(tag);
        self
    }
    
    /// Set URLs of pet photos
    pub fn with_photo_urls(mut self, photo_urls: String) -> Self {
        self.photo_urls = Some(photo_urls);
        self
    }
    
    
    
    /// Validate the instance
    pub fn validate(&self) -> Result<(), String> {
        // Add validation logic here if needed
        // name is required and present
        Ok(())
    }
}

impl Default for Newpet {
    fn default() -> Self {
        Self {
            name: Default::default(),
            tag: None,
            photo_urls: None,
        }
    }
}

impl std::fmt::Display for Newpet {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "Newpet \{ ")?;
        write!(f, "name: {:?}", self.name)?;
        write!(f, ", ")?;
        write!(f, "tag: {:?}", self.tag)?;
        write!(f, ", ")?;
        write!(f, "photo_urls: {:?}", self.photo_urls)?;
        write!(f, " \}")
    }
}